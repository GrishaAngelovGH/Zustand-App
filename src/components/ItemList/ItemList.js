import useStore from "hooks/useStore"

import Item from "components/Item"

const ItemList = () => {
  const items = useStore((state) => state.items)

  return (
    <div className="row g-0 mt-3 justify-content-evenly">
      {
        items.map(v => (
          <Item
            key={v.id}
            title={v.title}
            quantity={v.quantity}
            marked={v.marked}
          />
        ))
      }
    </div>
  )
}

export default ItemList