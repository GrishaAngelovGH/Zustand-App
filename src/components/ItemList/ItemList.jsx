import { useEffect } from "react"

import useStore from "hooks/useStore"

import Item from "components/Item"
import ItemStatus from "components/ItemStatus"

const isEmpty = obj => !Object.keys(obj).length

const ItemList = () => {
  const items = useStore((state) => state.items)
  const itemsStatus = useStore((state) => state.itemsStatus)
  const fetchItemsStatus = useStore((state) => state.fetchItemsStatus)
  const reset = useStore((state) => state.reset)

  useEffect(() => {
    fetchItemsStatus()
  }, [fetchItemsStatus, itemsStatus])

  return (
    <div className="flex-1 h-full">
      <div className="flex-col">
        <div className="flex mt-3 justify-evenly">
          {
            items.map(v => (
              <Item
                key={v.id}
                id={v.id}
                title={v.title}
                quantity={v.quantity}
                marked={v.marked}
              />
            ))
          }
        </div>

        <div className="flex justify-center mt-5">
          {
            isEmpty(itemsStatus) && (
              <div className="inline-block border-r-transparent rounded-full animate-spin border-4 border-blue-500 w-[5rem] h-[5rem]" role="status">
                <span className="hidden">Loading...</span>
              </div>
            )
          }
          {
            !isEmpty(itemsStatus) && (
              <div className="flex flex-col gap-5 items-center w-full">
                <div className="flex justify-around w-full">
                  {
                    items.map(item => (
                      <ItemStatus key={item.id} status={itemsStatus[item.id].status} />
                    ))
                  }
                </div>
                <button className="bg-red-500 text-white rounded-md w-[100px] mt-3 font-bold cursor-pointer p-2 text-2xl" onClick={reset}>Reset</button>
              </div>
            )
          }
        </div>
      </div>
    </div>
  )
}

export default ItemList