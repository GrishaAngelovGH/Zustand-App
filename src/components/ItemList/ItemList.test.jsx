import { render } from '@testing-library/react'

import ItemList from './ItemList'

test('should render ItemList component', () => {
  const view = render(<ItemList />)

  expect(view).toMatchSnapshot()
})