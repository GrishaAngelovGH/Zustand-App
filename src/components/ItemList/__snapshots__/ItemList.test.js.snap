// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ItemList component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="row g-0 mt-3 justify-content-evenly"
      >
        <div
          class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
        >
          <p
            class="fw-bold"
          >
            Title: 
            Item 1
          </p>
          <p
            class="fw-bold"
          >
            Quantity: 
            0
          </p>
          <p
            class="fw-bold"
          >
            Marked: 
            false
          </p>
        </div>
        <div
          class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
        >
          <p
            class="fw-bold"
          >
            Title: 
            Item 2
          </p>
          <p
            class="fw-bold"
          >
            Quantity: 
            0
          </p>
          <p
            class="fw-bold"
          >
            Marked: 
            false
          </p>
        </div>
        <div
          class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
        >
          <p
            class="fw-bold"
          >
            Title: 
            Item 3
          </p>
          <p
            class="fw-bold"
          >
            Quantity: 
            0
          </p>
          <p
            class="fw-bold"
          >
            Marked: 
            false
          </p>
        </div>
        <div
          class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
        >
          <p
            class="fw-bold"
          >
            Title: 
            Item 4
          </p>
          <p
            class="fw-bold"
          >
            Quantity: 
            0
          </p>
          <p
            class="fw-bold"
          >
            Marked: 
            false
          </p>
        </div>
        <div
          class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
        >
          <p
            class="fw-bold"
          >
            Title: 
            Item 5
          </p>
          <p
            class="fw-bold"
          >
            Quantity: 
            0
          </p>
          <p
            class="fw-bold"
          >
            Marked: 
            false
          </p>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="row g-0 mt-3 justify-content-evenly"
    >
      <div
        class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
      >
        <p
          class="fw-bold"
        >
          Title: 
          Item 1
        </p>
        <p
          class="fw-bold"
        >
          Quantity: 
          0
        </p>
        <p
          class="fw-bold"
        >
          Marked: 
          false
        </p>
      </div>
      <div
        class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
      >
        <p
          class="fw-bold"
        >
          Title: 
          Item 2
        </p>
        <p
          class="fw-bold"
        >
          Quantity: 
          0
        </p>
        <p
          class="fw-bold"
        >
          Marked: 
          false
        </p>
      </div>
      <div
        class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
      >
        <p
          class="fw-bold"
        >
          Title: 
          Item 3
        </p>
        <p
          class="fw-bold"
        >
          Quantity: 
          0
        </p>
        <p
          class="fw-bold"
        >
          Marked: 
          false
        </p>
      </div>
      <div
        class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
      >
        <p
          class="fw-bold"
        >
          Title: 
          Item 4
        </p>
        <p
          class="fw-bold"
        >
          Quantity: 
          0
        </p>
        <p
          class="fw-bold"
        >
          Marked: 
          false
        </p>
      </div>
      <div
        class="col-md-2 border border-3 border-primary rounded p-2 shadow bg-light"
      >
        <p
          class="fw-bold"
        >
          Title: 
          Item 5
        </p>
        <p
          class="fw-bold"
        >
          Quantity: 
          0
        </p>
        <p
          class="fw-bold"
        >
          Marked: 
          false
        </p>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
